src/memory/buffer_pool.cpp

```diff
@@ -6,30 +6,95 @@ SMILE_NS_BEGIN
 
 BufferPool::BufferPool( FileStorage* storage, const BufferPoolConfig& config ) noexcept {
 	p_storage = storage;
-	p_pool = (char*) malloc( config.m_poolSizeKB*1024*sizeof(char) );
+	m_config = config;	
+}
+
+ErrorCode BufferPool::open( const std::string& path ) noexcept {
+	ErrorCode error = p_storage->open(path);
+	if ( error != ErrorCode::E_NO_ERROR ) {
+		return error;
+	}
 
+	p_pool = (char*) malloc( m_config.m_poolSizeKB*1024*sizeof(char) );
```

Comment:
Its quite weird that you can specify a pool size which is not multiple of a page size. It is ok right now, but in the future we should issue a warning since it could be the consequence of an erro rwhen configuring the buffer pool.

Reply:
Fixed in 47abc39.